### PR TITLE
revisit: check HistoryReference exists before accessing it

### PR DESCRIPTION
**Pull request by drighty**
## Overview
` hr` may be null and causing `Revisit` fail

## Related Issues
Specify any related issues or pull requests by linking to them.

## Checklist
- [ ] Update help
- [ ] Update changelog
- [ ] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [ ] Sign-off commits
- [ ] Squash commits
- [ ] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by thc202** [comment]
- The changelog should be updated.
 - Would be good to add tests (though not a requirement given there's none yet).

File: addOns/revisit/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@ All notable changes to this add-on will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+## 5 - 2023-10-15
+### Changed
```
Comment (thc202) [inline addOns/revisit/CHANGELOG.md:7]: As mentioned in https://github.com/zaproxy/zap-extensions/blob/main/CONTRIBUTING.md#changelog this should be added to the Unreleased section. Should be under a `Fixed`.

**Comment by thc202** [comment]
Raised a new PR with this fix. Note that it's better to raise PRs from non-default branch as it causes issues when updating it.

**Comment by thc202** [comment]
Thank you for the fix!

**Comment by thc202** [comment]
@drighty how would you like to be credited (e.g. name, handle)?
https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team
